src/executor/evaluate/mod.rs

```diff
@@ -175,6 +179,51 @@ async fn evaluate_function<'a, T: 'static + Debug>(
 
             Ok(Evaluated::Value(value))
         }
+        name @ "LEFT" | name @ "RIGHT" => {
+            let return_wrap = |value| Ok(Evaluated::Value(Value::OptStr(value)));
```

Comment:
Oh, let's don't too rush

This code converts not only `Value::OptStr` to `OptStr`, but also it converts `Value::Str` to `Value::OptStr`

Reply:
I figured there wasn't much need for it, especially considering we'll be changing soon.

Do you think we need to return Str?

Reply:
This is not because of any special reason.
For example, this is the code above about `LOWER` and `UPPER`.
![image](https://user-images.githubusercontent.com/2025065/110561257-b65dc280-818a-11eb-993f-9cf7bffe010d.png)

LEFT | RIGHT of `Value::Str` becomes `Value::Str`.
LEFT | RIGHT of `Value::OptStr` becomes `Value::OptStr`.

Once, `LEFT` and `RIGHT` converts `Value::Str` to `Value::OptStr`, then `None` value of `Value::OptStr` becomes unreachable.
Using normal input sql, it is not possible to generate `None` from the result of `LEFT` + `Value::Str`.

Reply:
I suppose that's fair though it would require duplicating the code as the convert function takes two arguments.

Something to consider:
`LEFT` & `RIGHT` aren't just manipulations of a `Str`/`OptStr` column, they're their own values, determined by a `Str`/`OptStr` and `I64`/`OptI64`.

As per this, it *is* possible to generate `None` from the result of `LEFT` + `Str` **+** `OptI64(None)`.

Reply:
> As per this, it is possible to generate None from the result of LEFT + Str + OptI64(None).
Oh, thanks, then it make sense,

* LEFT("abcd", 2) => "ab"
* LEFT(NULL, 3) => NULL
* LEFT(NULL, NULL) => NULL
* LEFT("abcd", NULL) => NULL

I thought the result of `LEFT("abcd", NULL)` as "abcd".

`LEFT("abcd", NULL) => NULL` can convert `Str` to `OptStr`.
Ok, after you fix the small clippy issue and I'll merge this :) Thanks! 👍 👍